### DISCUSSION TITLE
[Chapter 10] Pub-Sub 메시징 기능에 필요한 네 가지 요소 중 토픽에 관하여

### DISCUSSION THREAD

**Discussion by corock** [cowaine/spring-boot-for-msa#170] [Chapter 10. Redis and Spring Cache] [answered]
책 10.7 레디스 Pub-Sub 구현 예제에서 그림 10-7 레디스의 Pub-Sub 기능을 설명할 때 아래와 같은 내용이 나옵니다.

> Pub-Sub 메시징 기능에는 네 가지 요소가 필요하다.
>
> 서버 사이에 전달되는 데이터인 메시지,
> 메시지 데이터를 생산하는 게시자(_publisher_),
> 생산된 이벤트를 수신하는 구독자(_subscriber_),
> 이들 사이에서 메시지를 큐잉하고 구독자가 메시지를 수신할 수 있는 토픽(_topic_)이다.

근데 그림을 보면 토픽 역할을 하는 녀석이 문제가 생기면 어떻게 되는 건가? 를 생각해 봤습니다.
큐잉만 된 채로 메시지 전달이 안 되고 장애가 발생할 것 같은데, 이런 경우도 있는지와 있다면 어떻게 대처해야 할까요?

**Comment by JoisFe** [comment]
topic이 RDB로 따지면 table로 볼 수 있따고 하던데, 잘 못된 table에 데이터를 insert 하면 이상하지 않을까?

**Comment by corock** [comment]
cf. https://oliveyoung.tech/blog/2023-08-07/async-process-of-coupon-issuance-using-redis/

**Comment by Irisation23** [comment]
### 금주 Check Point

- 카프카에서는 다음과 같이 마스터-슬레이브 구조의 파티션을 두고 있음
<img width="248" alt="image" src="https://github.com/cowaine/spring-boot-for-msa/assets/53285909/9b8f64aa-c968-45fe-a3b3-e65b6e613bcf">

**Comment by JoisFe** [comment, marked as answer]
https://www.youtube.com/watch?v=uk5fRLUsBfk&list=PL42XJKPNDepYXyKefvicxlA2fz1aThVs5&index=9
